File: backend/api/Models/Recipe.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using api.Enums;

namespace api.Models
{
    public class Recipe
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public CostType CostType { get; set; }
        public RecipeType RecipeType { get; set; }
        public string CreatedByUserId { get; set; } = string.Empty;
        public decimal? AverageRating { get; set; }
        public string? ImageUrl { get; set; }
        public int RatingCount { get; set; }
        public decimal BayesianScore { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public ICollection<RecipeInteraction> Interactions { get; set; } = new List<RecipeInteraction>();

    }
}

File: frontend/src/types/recipe.ts
export const RecipeType = {
  Meat: 1,
  Fish: 2,
  Vegetarian: 3,
  Vegan: 4,
} as const;

export type RecipeType = (typeof RecipeType)[keyof typeof RecipeType];

export const CostType = {
  Cheap: 1,
  Medium: 2,
  Expensive: 3,
} as const;

export type CostType = (typeof CostType)[keyof typeof CostType];

export type Recipe = {
  id: number;
  name: string;
  description: string;
  imageUrl?: string | null; 
  recipeType: RecipeType;
  costType: CostType;
  averageRating: number | null;
  ratingCount: number;
  bayesianScore: number;
};


